rasa/utils/train_utils.py

```diff
@@ -531,6 +533,21 @@ def validate_configuration_settings(component_config: Dict[Text, Any]) -> None:
     _check_loss_setting(component_config)
     _check_confidence_setting(component_config)
     _check_similarity_loss_setting(component_config)
+    _check_checkpoint_setting(component_config)
+
+
+def _check_checkpoint_setting(component_config: Dict[Text, Any]) -> None:
+    if component_config[CHECKPOINT_MODEL]:
+        if component_config[EVAL_NUM_EPOCHS] <= 0:
+            rasa.shared.utils.io.raise_warning(
+                f"You have opted to save the best model, {EVAL_NUM_EPOCHS} is not greater "
+                f"than 0, training will fail."
+            )
+        if component_config[EVAL_NUM_EXAMPLES] <= 0:
+            rasa.shared.utils.io.raise_warning(
+                f"You have opted to save the best model, {EVAL_NUM_EXAMPLES} is not greater "
+                f"than 0. No model will be saved."
+            )
```

Comment:
Eek, actually just realised that `EVAL_NUM_EPOCHS` is allowed to be `-1`... Sorry for this confusion, will address that.

Not sure what you mean by enable/disable validation. Really the correct way to disable is to set `checkpoint_model` to false. Should I add information about the consequences of incorrect configurations?

Reply:
> Not sure what you mean by enable/disable validation. Really the correct way to disable is to set checkpoint_model to false.

What I mean is enabling/disabling model validation. Let's consider this comment that accompanies `EVAL_NUM_EPOCHS` in TED: `How often calculate validation accuracy. Small values may hurt performance, e.g. model accuracy.` For now ignoring the part about hurting model accuracy (which I kinda don't get), I might want to disable validation altogether if I want to train faster. And I might be completely unaware of model checkpointing. In such case, it'd help me to know how to control validation as such.

With `EVAL_NUM_EPOCHS` being possible to set to -1, this becomes easier indeed, but I still have to figure out this magical special value of `-1`. I think that a simple comment 1) in all the models where `EVAL_NUM_EPOCHS` is listed as a parameter and 2) in the docs (in a visible way) would help here. I'm unsure if it's still within the scope of this issue though.

Reply:
Ohh I see, you mean enable/disable validation separately from enable/disable saving the model. Sorry, I've been (incorrectly) conflating the two.

I actually don't think this can be done with the `EVAL_NUM_EPOCHS` param, because even `EVAL_NUM_EPOCHS=-1` will still validate at the very end of training. You'd have to set `EVAL_NUM_EXAMPLES=0` for no validation. I've adjusted the comments in the code and tried to clarify in the docs.

Reply:
Makes sense. Thanks!